src/backend/distributed/transaction/relation_access_tracking.c

```diff
@@ -281,11 +284,68 @@ RecordPlacementAccessToCache(Oid relationId, ShardPlacementAccessType accessType
 }
 
 
+/*
+ * RecordParallelAccessForTaskList gets a task list and records
+ * the necessary parallel relation accesses for the task list.
+ *
+ * This function is used to enforce foreign keys from distributed
+ * tables to reference tables.
+ */
+void
+RecordParallelAccessForTaskList(List *taskList)
```

Comment:
As @marcocitus noted in #2692 we should change the name of the function

> I think keeping the RecordParallel -> RecordRelationParallel convention would be nice. Then it's "obvious" why we only care about the first task.

Reply:
Some comment explaining that all tasks contains the same set of relations, so we are only processing the first task, can be useful.